wgpu-core/src/track/metadata.rs

```diff
@@ -17,25 +13,25 @@ use wgt::strict_assert;
 /// members, but a bit vector tracks occupancy, so iteration touches
 /// only occupied elements.
 #[derive(Debug)]
-pub(super) struct ResourceMetadata<A: HalApi> {
+pub(super) struct ResourceMetadata<A: HalApi, I: TypedId, T: Resource<I>> {
     /// If the resource with index `i` is a member, `owned[i]` is `true`.
     owned: BitVec<usize>,
 
     /// A vector parallel to `owned`, holding clones of members' `RefCount`s.
-    ref_counts: Vec<Option<RefCount>>,
+    resources: Vec<Option<Arc<T>>>,
 
     /// A vector parallel to `owned`, holding the epoch of each members' id.
     epochs: Vec<Epoch>,
```

Comment:
There's no longer any need for this field; see #3803.

Reply:
Oki I see, we can get rid of It.
Do we need anyway the epoch in the id and when unzipping it, then?

Reply:
Maybe I'm misunderstanding your question, but we definitely want to keep the epoch in the id; that's essential to the checks in `Storage::get`, `Storage::try_get`, and so on. But #3803 (the [second-to-last commit](https://github.com/gfx-rs/wgpu/pull/3803/commits/3bb4f417502cadbbf691e3bcb47e2030711fd7aa), at least) shows that it's not needed much in `wgpu-core/src/track/*`.